pygmt/src/solar.py

```diff
@@ -0,0 +1,97 @@
+"""
+solar - Plot day-night terminators.
+"""
+
+import datetime
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_string, fmt_docstring, kwargs_to_strings, use_alias
+
+
+@fmt_docstring
+@use_alias(
+    R="region",
+    J="projection",
+    G="fill",
+    B="frame",
+    W="pen",
+    U="timestamp",
+    V="verbose",
+    X="xshift",
+    Y="yshift",
+    c="panel",
+    p="perspective",
+    t="transparency",
+)
+@kwargs_to_strings(R="sequence", p="sequence")
+def solar(self, terminator="d", terminator_datetime=None, **kwargs):
+    r"""
+    Plot day-light terminators.
+
+    This function plots the day-night terminator. Additionally, it can
+    plot the terminators for civil twilight, nautical twilight, and
+    astronomical twilight.
+
```

Comment:
Add link to gmt docs.

```suggestion

    Full parameter list at :gmt-docs:`solar.html`
    
```